ros2doctor/ros2doctor/api/__init__.py

```diff
@@ -18,16 +18,23 @@
 def run_checks():
     """Run all checks when `ros2 doctor/wtf` is called."""
     all_results = []
-    failed_names = []
+    failed_checks = []
+    failed_modules = []
     for check in iter_entry_points('ros2doctor.checks'):
         result = check.load()()  # load() returns method
         all_results.append(result)
         if result is False:
-            failed_names.append(check.name)
-    return all_results, failed_names
+            failed_checks.append(check.name)
+            failed_modules.append(check.module_name)
+    return all_results, failed_checks, failed_modules
 
 
 def generate_report():
     """Print report to terminal when `-r/--report` is attached."""
-    for report in iter_entry_points('ros2doctor.report'):
-        report.load()()  # load() returns method
+    report = {}
+    for r in iter_entry_points('ros2doctor.report'):
+        if r.module_name in report:
+            report[r.module_name].extend(r.load()())
```

Comment:
Recommend using something other than the module name since reports in the same module could check very different things. Maybe the name of the entry point?

```python3
report[r.name] = r.load()()
```


```Python3
>>> list(pr.iter_entry_points('ros2doctor.report'))[0]
EntryPoint.parse('report_network_config = ros2doctor.api.network:print_network')
>>> list(pr.iter_entry_points('ros2doctor.report'))[0].module_name
'ros2doctor.api.network'
>>> list(pr.iter_entry_points('ros2doctor.report'))[0].name
'report_network_config'
```

Reply:
`module_name` is the only direct correlation between checks and reports at this point. I use it as key to report any failed check. The design goal of this part is to make check and report inside the same module serve one specific area like `platform`, `network`, `topic` etc. I can see when there could be a lot of checks inside one module, but to tailor each report around a single check would require some string manipulation and strict rules on report function names.